src/ItemMeasurer.js

```diff
@@ -0,0 +1,107 @@
+// @flow
+
+import { Component } from 'react';
+import { findDOMNode } from 'react-dom';
+
+import type { Direction } from './createListComponent';
+import type { HandleNewMeasurements } from './DynamicSizeList';
+
+class DOMRectReadOnly {
+  +x: number;
+  +y: number;
+  +width: number;
+  +height: number;
+  +top: number;
+  +right: number;
+  +bottom: number;
+  +left: number;
+}
+
+class ResizeObserverEntry {
+  +target: HTMLElement;
+  +contentRect: DOMRectReadOnly;
+}
+
+type Entries = $ReadOnlyArray<ResizeObserverEntry>;
+
+type ResizeObserverCallback = {
+  // eslint-disable-next-line no-use-before-define
+  (entries: Entries, observer: ResizeObserver): void,
+};
+
+declare class ResizeObserver {
+  constructor(ResizeObserverCallback): ResizeObserver;
+  observe(target: HTMLElement): void;
+  unobserve(target: HTMLElement): void;
+  disconnect(): void;
+}
+
+type ItemMeasurerProps = {|
+  direction: Direction,
+  handleNewMeasurements: HandleNewMeasurements,
+  index: number,
+  item: React$Element<any>,
+  size: number,
+|};
+
+export default class ItemMeasurer extends Component<ItemMeasurerProps, void> {
+  _node: HTMLElement | null = null;
+  _resizeObserver: ResizeObserver | null = null;
+
+  componentDidMount() {
+    const node = ((findDOMNode(this): any): HTMLElement);
+    this._node = node;
+
+    // Force sync measure for the initial mount.
+    // This is necessary to support the DynamicSizeList layout logic.
+    this._measureItem(true);
+
+    if (typeof ResizeObserver !== 'undefined') {
+      // Watch for resizes due to changed content,
+      // Or changes in the size of the parent container.
+      this._resizeObserver = new ResizeObserver(this._onResize);
```

Comment:
I use ponyfill and would like to pass this class via props. Are you good with this?